src/scenarios.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+import csv
+
+import numpy as np
+from itertools import product
+
+from sim import simulate as sim
+
+#TODO add charger_density once supported in sim.py
+def genScenarios(battery_cap_kwh,nondepot_charger_rate,): #charger_density):
+    '''Generates a list of tuples of scenarios
+        args: all tuples of a start (inclusive), end (exclusive), and step 
+        size values. i.e. battery_cap_kwh=(125,700,100) 
+    '''
+    #TODO include error handling
+    bat_range = np.arange(*battery_cap_kwh)
+    non_depot_range = np.arange(*nondepot_charger_rate)
+    #charger_density = np.arange(*charger_density)
+
+    scen_opts = [bat_range,non_depot_range,]# charger_density]
+    scens = list(product(*scen_opts))
+    return scens
+
+def main():
+  parser = argparse.ArgumentParser(description='TODO')
+  parser.add_argument('parsed_gtfs_prefix', type=str, help='TODO')
+  parser.add_argument('osm_data',           type=str, help='TODO')
+  parser.add_argument('depots_filename',    type=str, help='TODO')
+  parser.add_argument('output_filename',    type=str, help='TODO')
+  parser.add_argument('-b','--battery-cap-kwh',      nargs=3,type=int,help='TODO')
+  parser.add_argument('-c','--nondepot-charger-rate',nargs=3,type=int,help='TODO')
```

Comment:
Rather than taking in `nargs`, perhaps they could take a comma-delimited list of numbers and internally separate those?

Reply:
could you elaborate? Where would we separate? after parsing the args? Why is the advantageous over ensuring 3 values get passed?

Reply:
You'd probably separate after parsing the arguments by overwriting the args value with a list. The advantage is that it allows you to take variable-length lists without having to modify the code.

Reply:
humm, well if you dont pass exactly 3 arguments its going to blow up either here or on `genScenarios()`. Is there a best practice for where that error handling happen? Both? just genScenarios? or as we have it here at the parser? 

Thinking aloud, so to speak. I guess if you are using this as a module than it would make sense to do so at the `genScenarios()` level.